git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,620 @@
+
+
+![The Commit Stack: 3 flat white boxes, stacked on top of each other. The top box is labeled "last committed version of repository", the middle box is labeled "next to last version of repository" and the bottom is labeled "previous version of repository". Three dots below the lowest box indicate that this pattern continues.](media/Commit_stack.svg)
+
+### Using HEAD
+
+The `HEAD` refers to the most recently commit to your repository. It won't include any changes that haven't been committed to the repository yet. You can see that most recent commit by entering `git show HEAD`.
+
+```
+$ git show HEAD
+commit 584977a18319f539f3f5ecde6176db2ba0081cd9 (HEAD -> main, origin/main)
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Mon Mar 14 14:59:40 2022 -0400
+
+    Ignore data files and the results folder.
+
+diff --git a/.gitignore b/.gitignore
+new file mode 100644
+index 0000000..f451386
+--- /dev/null
++++ b/.gitignore
+@@ -0,0 +1,2 @@
++*.dat
++results/
+```
+
+The last thing that Dracula committed to the planets repository was the `.gitignore` file in which he told Git to "Ignore data files and the results folder."
+
+<div class = "care">
+Sometimes a command will output many lines of code that may not be relevant to you. For example the `show` command will output not just the commit message, but also every single change that you made in that commit.
+
+The more you see this type of output, the more comfortable you will get with extracting the information you need and ignoring the rest.
+</div>
+
+<div class = "important">
+If the output of a command is longer than the number of lines your console displays, you can navigate that output using the down and up arrows on your keyboard, or press `q` to skip to the end of that output.
+</div>
+
+Maybe you want to look one step further back into your work. By using `HEAD~n` you can look back n checkpoints in your repository. The `~` symbol is named "tilde" and pronounced "TIL-duh."
+
+For example, to look back just one commit before the most recent checkpoint, use `git show HEAD~1`. If instead you wanted to look back three checkpoints (two commits before the most recent commit) you would enter `git show HEAD~2`.
+
+![The stack of 3 white flat boxes. Now the top box is labeled `HEAD`, the middle box `HEAD~1` and the bottom `HEAD~2`](media/Commit_stack_HEAD.svg)
+
+Using `HEAD` to refer to your commits can be great for looking at recent versions of your repository. But be careful, each time you commit to the repository, you are creating a new most-recent-version that is now the `HEAD` and every older version moves one layer down in the stack.
+
+![On the left, the stack of 3 flat white boxes are labeled `HEAD`, `HEAD~1`,  and `HEAD~2`. An arrow points to a stack on the right made of the 3 white boxes with a 4th red box now on top. This red box is labeled `HEAD` and the boxes below it are labeled, in order from top to bottom, `HEAD~1`, `HEAD~2`, `HEAD~3`](media/Commit_stack_HEAD2.svg)
```

Comment:
```suggestion
![On the left, the stack of 3 flat white boxes are labeled `HEAD`, `HEAD~1`,  and `HEAD~2`. An arrow points to a stack on the right made of the 3 white boxes with a 4th red box now on top. This red box is now the one labeled `HEAD` and the boxes below it are labeled, in order from top to bottom, `HEAD~1`, `HEAD~2`, `HEAD~3`, with all the white boxes having changed labels.](media/Commit_stack_HEAD2.svg)
```

I'm not great at alt texts so feel free to veto the suggestion.  ALSO, time saver (but no need to change it)... you can just put backticks around the entire alt text instead of doing on individual words... the output is the same.